find.go

```diff
@@ -29,26 +31,52 @@ type UPNPDevice struct {
 	Desc    *deviceDesc
 }
 
-// FindHost looks for a host matching the query, returns just the host/ip (no port).
-func FindHost(ctx context.Context, logger logging.Logger, query DeviceQuery) (string, error) {
-	all, err := findAll(ctx, logger, query.Network)
+func parseNetworks(queries []DeviceQuery) []string {
+	networks := []string{}
+	for _, query := range queries {
+		if query.Network != "" && slices.Contains(networks, query.Network) {
+			networks = append(networks, query.Network)
+		}
+	}
+	return networks
+}
+
+// FindHost looks for a host matching the query, returns the host/ip (no port).
```

Comment:
need to make clear if queries is an OR or AND

Reply:
updated the comment. Assuming you meant to mention how `matching` worked so I added a bit for that